ivadomed/transforms.py

```diff
@@ -1,26 +1,44 @@
+
 import math
 import numbers
 import random
+import functools
 import numpy as np
 from PIL import Image
 
 from skimage.exposure import equalize_adapthist
+from skimage.transform import resize
 from scipy.ndimage.measurements import label, center_of_mass
 from scipy.ndimage.filters import gaussian_filter
 from scipy.ndimage.morphology import binary_dilation, binary_fill_holes, binary_closing
 from scipy.ndimage.interpolation import map_coordinates
 
 import torch
-import torchvision.transforms.functional as F
-from torchvision import transforms as torchvision_transforms
+#import torchvision.transforms.functional as F
+#from torchvision import transforms as torchvision_transforms
+
+
+def list_capable(wrapped):
+    @functools.wraps(wrapped)
+    def wrapper(self, sample, metadata):
+        if isinstance(sample, list):
+            list_data, list_metadata = [], []
+            for s_cur, m_cur in zip(sample, metadata):
+                # Run function for each sample of the list
+                data_cur, metadata_cur = wrapped(self, s_cur, m_cur)
+                list_data.append(data_cur)
+                list_metadata.append(metadata_cur)
+            return list_data, list_metadata
+        return wrapped(self, sample, metadata)
+    return wrapper
 
 
 class IMEDTransform(object):
```

Comment:
Maybe use the camel convention? 'ImedTransform'

Reply:
Done :)